packages/client-nestjs/src/client-worker.integration.spec.ts

```diff
@@ -0,0 +1,273 @@
+/* eslint-disable sort-imports -- Integration test imports order */
+import {
+  defineConsumer,
+  defineContract,
+  defineExchange,
+  defineMessage,
+  definePublisher,
+  defineQueue,
+  defineQueueBinding,
+} from "@amqp-contract/contract";
+import { AmqpClientModule } from "./client.module.js";
+import { AmqpClientService } from "./client.service.js";
+import { Test } from "@nestjs/testing";
+import { describe, expect, vi } from "vitest";
+import { it as baseIt } from "@amqp-contract/testing/extension";
+import { z } from "zod";
+
+// Define contract at module level for type inference
+const ordersExchange = defineExchange("orders-exchange", "topic", { durable: false });
+const ordersQueue = defineQueue("orders-queue", { durable: false });
+const ordersBinding = defineQueueBinding(ordersQueue, ordersExchange, {
+  routingKey: "order.#",
+});
+
+const orderMessage = defineMessage(
+  z.object({
+    orderId: z.string(),
+    customerId: z.string(),
+    amount: z.number(),
+  }),
+);
+
+const testContract = defineContract({
+  exchanges: {
+    orders: ordersExchange,
+  },
+  queues: {
+    orders: ordersQueue,
+  },
+  bindings: {
+    ordersBinding,
+  },
+  publishers: {
+    orderCreated: definePublisher(ordersExchange, orderMessage, {
+      routingKey: "order.created",
+    }),
+  },
+  consumers: {
+    processOrder: defineConsumer(ordersQueue, orderMessage),
+  },
+});
+
+const it = baseIt;
+
+describe("Client and Worker Integration", () => {
+  describe("end-to-end message flow", () => {
+    it("should publish from client and consume in worker", async ({ amqpConnectionUrl }) => {
+      // GIVEN - handler mock to track consumed messages
+      const handler = vi.fn().mockResolvedValue();
+
+      // Import worker module dynamically to avoid circular dependency issues
+      const { AmqpWorkerModule } = await import("../../worker-nestjs/src/worker.module.js");
+
+      // Create both client and worker modules
+      const moduleRef = await Test.createTestingModule({
+        imports: [
+          AmqpClientModule.forRoot({
+            contract: testContract,
+            urls: [amqpConnectionUrl],
+          }),
```

Comment:
no `AmqpClientModule` when we test AmqpWorkerModule and vice versa. otherwise, when tests are ko we are to able to tell if the issue is with worker or client